src/engine/parser/error.rs

```diff
@@ -0,0 +1,190 @@
+// This file is part of rust-web/twig
+//
+// For the copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+//! Typisation of parser errors.
+
+use std::fmt::{self, Display};
+use error::Error;
+use error::{GeneralizeTo, ErrorCode};
+
+use engine::parser::token::TokenErrorCode;
+use engine::parser::job::{self, cursor};
+use engine::parser::token;
+
+pub type ParserError = Error<ParserErrorCode>;
+pub type NodeError = Error<NodeErrorCode>; // todo move somewhere else??
+
+impl GeneralizeTo<ParserErrorCode> for TokenErrorCode {
+    fn generalize(&self) -> ParserErrorCode { ParserErrorCode::TokenError }
+}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub enum ParserErrorCode {
+    Unreachable {
+        reason: String,
+        job: job::JobDump,
+    },
+    MissingExtensions,
+    InvalidState {
+        item: token::stream::Item,
+        job: job::JobDump,
+    },
+    TokenError,
```

Comment:
More general `TokenError` in the same list as more specific `TokenParserError`. I think both can be specific?